Source/WebKit/UIProcess/WebPageProxy.h

```diff
@@ -2781,6 +2783,41 @@ class WebPageProxy final : public API::ObjectImpl<API::Object::Type::Page>, publ
     Internals& internals() { return m_internals; }
     const Internals& internals() const { return m_internals; }
 
+    class ProcessActivityState {
+        WebPageProxy& m_page;
+
+        std::unique_ptr<ProcessThrottlerActivity> m_isVisibleActivity;
+#if PLATFORM(MAC)
+        ProcessThrottlerTimedActivity m_wasRecentlyVisibleActivity;
+#endif
+        std::unique_ptr<ProcessThrottlerActivity> m_isAudibleActivity;
+        std::unique_ptr<ProcessThrottlerActivity> m_isCapturingActivity;
+#if PLATFORM(IOS_FAMILY)
+        std::unique_ptr<ProcessThrottlerActivity> m_openingAppLinkActivity;
+#endif
+
+    public:
```

Comment:
These should come first and data members last. Please follow the style of the WebKit code base.